src/services/cardano-register-handler/cardano-register-handler.service.ts

```diff
@@ -168,21 +132,14 @@ export async function registerAgentV1() {
                 other: stringToMetadata(request.other),
               },
               tags: request.tags,
-              agentPricing:
-                request.Pricing.pricingType == PricingType.Fixed
-                  ? {
-                      pricingType: PricingType.Fixed,
-                      fixedPricing:
-                        request.Pricing.FixedPricing?.Amounts.map(
-                          (pricing) => ({
-                            unit: stringToMetadata(pricing.unit),
-                            amount: pricing.amount.toString(),
-                          }),
-                        ) ?? [],
-                    }
-                  : {
-                      pricingType: PricingType.Free,
-                    },
+              agentPricing: {
+                pricingType: request.Pricing.pricingType,
+                fixedPricing:
+                  request.Pricing.FixedPricing?.Amounts.map((pricing) => ({
+                    unit: stringToMetadata(pricing.unit),
+                    amount: pricing.amount.toString(),
+                  })) ?? [],
```

Comment:
Also changes the logic, as otherwise it is expected to be empty not checked, here its set as an empty array